pdl-compiler/tests/canonical/le_test_file.pdl

```diff
@@ -362,6 +362,26 @@ packet Packet_Array_Field_UnsizedElement_VariableCount_Padded {
     _padding_ [16],
 }
 
+struct Packet_Array_ElementSize_Struct_ {
```

Comment:
I feel we need these tests, to keep exhaustiveness:
```
packet Packet_Array_VariableElementSize_ConstantSize {}
packet Packet_Array_VariableElementSize_VariableSize {}
packet Packet_Array_VariableElementSize_VariableCount {}
packet Packet_Array_VariableElementSize_UnknownSize {}
```
with more variations for `Struct`, `SizedCustomField` (useless though it is), `UnsizedCustomField`.
and I know that some of those you won't support ha ha :imp: